mmcif/io/BinaryCifWriter.py

```
@@ -113,14 +113,16 @@ def __encodeColumnData(self, colDataList, dataType):
         colMaskDict = None  # Use None when no mask and not {} - per Mol* implementation
         enc = BinaryCifEncoders(defaultStringEncoding=self.__defaultStringEncoding, storeStringsAsBytes=self.__storeStringsAsBytes, useFloat64=self.__useFloat64)
         #
-        maskEncoderList = ["Delta", "RunLength", "IntegerPacking", "ByteArray"]
+        maskEncoderList = ["RunLength", "ByteArray"]
         typeEncoderD = {"string": "StringArrayMasked", "integer": "IntArrayMasked", "float": "FloatArrayMasked"}
         colMaskList = enc.getMask(colDataList)
         dataEncType = typeEncoderD[dataType]
         colDataEncoded, colDataEncodingDictL = enc.encodeWithMask(colDataList, colMaskList, dataEncType)
         if colMaskList:
-            maskEncoded, maskEncodingDictL = enc.encode(colMaskList, maskEncoderList, "integer")
-            colMaskDict = {self.__toBytes("data"): maskEncoded, self.__toBytes("encoding"): maskEncodingDictL}
+            # Mol* indicates that masks should be encoded as if uint_8
+            colMaskListTyped = TypedArray(colMaskList, "unsigned_integer_8")
+            maskEncoded, maskEncodingDictL = enc.encode(colMaskListTyped, maskEncoderList, "integer")
+            colMaskDict = {self.__toBytes("data"): maskEncoded.data, self.__toBytes("encoding"): maskEncodingDictL}
```

Was the use of `maskEncoded` instead of `maskEncoded.data` an oversight previously? Or is it because the `.encode()` method is being provided `colMaskListTyped` now instead of `colMaskList`?

---

This is an artifact of John W/ original implementation.  When I added the TypedArray - I "overloaded" the encode method to handle if the data coming in was a TypedArray or a list.  If a list came in, a list would go out.

As a number of the interfaces are public ("integerPackingEncoder") - I added typed varieties - with deprecation warnings for 2026.

encode() will only invoke the typed encoding methods now. 

So - the work with Sebastian - I never changed over the mask code to use typed arrays. Through this ticket - I discovered that Mol* was using an unsigned int 8 bit array for the mask - so I did the same.

---

I see that there are still some places we are not using type arrays --- in the string encodings offset list and index list, and then float array masked list.  These should be addressed separately.  (we likely do not have instances masks for floating arrays as these tend to be occupancy, B factors, x, y, z which are likey all present - but with the mol* type hinting - we should address).

The original issue here for this was the default value of 0 in the array vs -1 -- but fixing the integer mask was part of debugging the problem.